account_invoice_import_invoice2data/tests/test_invoice2data_template.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# © 2017 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp.tests.common import SavepointCase
+import base64
+from openerp.tools import file_open, float_compare, config
+
+
+class TestInvoice2dataTemplate(SavepointCase):
+
+    def setUp(cls):
+        super(TestInvoice2dataTemplate, cls).setUp()
+        cls.templates = cls.env['invoice2data.template']
+
+    def test_00_invoice2data_template(self):
+        self.assertEquals(0, len(self.templates.get_templates('pi')))
+        self._create_template()
+        self.assertEquals(1, len(self.templates.get_templates('pi')))
+
+    def test_01_import_custom_invoice_template(self):
+        config.__setitem__('invoice2data_exclude_built_in_templates', True)
+        # Be sure we exclude built in templates
+        self.assertTrue(config.get('invoice2data_exclude_built_in_templates'))
```

Comment:
Because the contents of the database template is equal to the contents of the file system template, this test does not demonstrate that the new functionality actually works. Maybe if you switch the date_due and the date around you can prove that it is the database template that is being used because the dates in the resulting Odoo invoice are switched around accordingly

Reply:
I've extended the test by switching dates. There seems to be an issue with the attached pdf for the existing test (i guess that's the reason why it is commented out)

Reply:
The original test is also fixed